org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/semantictokens/TokenModifier.java

```diff
@@ -31,54 +31,26 @@
 import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
 
 public enum TokenModifier {
-	STATIC("static") {
-		@Override
-		protected boolean applies(IBinding binding) {
-			return Modifier.isStatic(binding.getModifiers());
-		}
-	},
-	FINAL("readonly") {
-		@Override
-		protected boolean applies(IBinding binding) {
-			return Modifier.isFinal(binding.getModifiers());
-		}
-	},
-	DEPRECATED("deprecated") {
-		@Override
-		protected boolean applies(IBinding binding) {
-			return binding.isDeprecated();
-		}
-	},
-	PUBLIC("public") {
-		@Override
-		protected boolean applies(IBinding binding) {
-			return Modifier.isPublic(binding.getModifiers());
-		}
-	},
-	PRIVATE("private") {
-		@Override
-		protected boolean applies(IBinding binding) {
-			return Modifier.isPrivate(binding.getModifiers());
-		}
-	},
-	PROTECTED("protected") {
-		@Override
-		protected boolean applies(IBinding binding) {
-			return Modifier.isProtected(binding.getModifiers());
-		}
-	},
-	ABSTRACT("abstract") {
-		@Override
-		protected boolean applies(IBinding binding) {
-			return Modifier.isAbstract(binding.getModifiers());
-		}
-	},
-	DECLARATION("declaration") {
-		@Override
-		protected boolean applies(SimpleName simpleName) {
-			return isDeclaration(simpleName);
-		}
-	};
+	// Standard Java modifiers
+	PUBLIC("public"),
```

Comment:
I'm wondering whether we have to add **all** "standard Java modifiers", because ultimately we'll align modifiers with [LSP3.16](https://microsoft.github.io/language-server-protocol/specifications/specification-3-16/#textDocument_semanticTokens) as much as possible, via LSP4J. 

And massive modifiers would be potentially anonying (we did receive such feedback). I'm thinking, do people actually want to know below information from highlighting?
```
transient, volatile, default, synchronized, native, strictfp
```

My suggestion is to only keep important modifiers you are actually consuming now, also to avoid future breaking changes (when aligning to LSP) as best as we can.

Reply:
Well, I do see your point here. The main purpose of this PR was to add the `generic`, `typeArgument` and `importDeclaration` modifiers. But I realised that adding the rest of the standard Java modifiers was easy and computationally cheap, so I thought "why not?". 

I'm not sure I understand what you mean by "align modifiers with LSP3.16"? We still need to have some custom modifiers that are not part of LSP3.16, and I don't see why these would be any different. Regarding the "massive modifiers", I think it would be rare to actually see all of those modifiers used together in a token, since many of them are mutually exclusive, or at least rarely used together. 

To be clear, I also think some of the standard Java modifiers are rarely going to be used, like `strictfp` for example. But I just didn't see any particular reason not to add the information, since it's trivial to check them in the same way as `public`, `static`, `readonly` etc. If you want me to remove the new "standard Java modifiers", I am also fine with that.

Reply:
I'm not that strongly opposed to that (as you mentioned you don't see any slow-down), my point is just "why to add them if they are rarely going to be used". It's indeed trivial to check them, it may cost tiny but still some time to check...

I still prefer not to add them here. One day if we do consume those modifiers, it's also trivial to add them back.

Reply:
Yeah, I do see your point. I'll remove those modifiers from this PR in my next iteration, which I'll probably commit later this week. But can we keep the `native` modifier? I don't see much use for the other modifiers (`transient`, `volatile`, `default`, `synchronized` and `strictfp`), but `native` is something that I might actually use myself, as it's quite common to see when using native libraries.